pkg/build/api/v1/types.go

```diff
@@ -303,6 +303,9 @@ type BuildStrategy struct {
 
 	// CustomStrategy holds the parameters to the Custom build strategy
 	CustomStrategy *CustomBuildStrategy `json:"customStrategy,omitempty" description:"holds parameters to the Custom build strategy"`
+
+	// ExternalStrategy holds the parameters to the External build strategy
+	ExternalStrategy *ExternalBuildStrategy `json:"externalStrategy,omitempty" description:"holds parameters to the External build strategy"`
```

Comment:
we can get the Jenkins plugin to write the status of each Jenkins pipeline build back into the OpenShift Build object with details of what stages have completed, what are pending & the expected duration of each stage; so that the console or CLI can visualise pipeline builds purely using the metadata inside the OpenShift Build object
https://github.com/fabric8io/openshift-jenkins-sync-plugin/issues/2